components/src/core/utility/spacer.tsx

```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+import { StyleProp, StyleSheet, View, ViewStyle, ViewProps } from 'react-native';
+
+const spacerStyles = (props: SpacerProps): StyleSheet.NamedStyles<{ root: ViewStyle }> =>
+    StyleSheet.create({
+        root: {
+            flexGrow: props.flex,
+            flexShrink: props.flex,
+            flexBasis: props.flex === 0 ? 'auto' : 0,
```

Comment:
Hmm... React's implementation is 
```js
flex: `${flex} ${flex} ${flex === 0 ? 'auto' : '0px'}`
```

And MDN suggest that when you are using one-argument syntax, it is interpreted as 
```js
flex: `${flex} 1 0`
```

| flex | 0 | 1|
| -|-|-|
|React|0 0 auto| 1 1 0px|
|Your suggestion|0 1 0|1 1 0|

Should I change the spacer implementation in React as well?

Reply:
Flex works slightly differently in RN than in web: https://reactnative.dev/docs/layout-props#flex

So my proposal would be:
| flex | 0 | 1| 2|
| -|-|-|-|
|React|0 0 auto| 1 1 0px| 2 2 0px |
|Your suggestion|0 0 auto|1 1 0| 2 1 0|

I don't propose changing anything in React/Web. That's exactly how we want it to behave. I'm proposing this approach for RN because it lines up better with the typical usage of flex (single number):
```
<View style={{flex: 1}}/>
<View style={{flex: 3}}/>
<View style={{flex: 1}}/>
```

The flex-shrink value in this case would be 1 for all three `<View>`s. If we are changing the flex-shrink property for the `<Spacer>` component, but the typical usage for other components gives them one...the results may be different than expected for smaller containers (things shrinking at different rates).

Reply:
In practice it will probably make very little difference either way...in most cases if somebody is running into a situation on mobile where they are encountering flex-shrink, they're probably going to design the screen differently to avoid it (i.e., scroll). I was just opting for the approach that writes less code 😄